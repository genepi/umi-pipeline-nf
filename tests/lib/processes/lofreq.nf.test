nextflow_process {

    name "Test Process LOFREQ"
    script "lib/processes/lofreq.nf"
    process "LOFREQ"

    test("Should run without failures") {

        when {
            params {
                variant_calling = true 
                variant_caller = "lofreq"
            }
            process {
                """
                input[0] = Channel.of(["barcode03", "final", "$baseDir/tests/input/variant_calling/barcode03/final.bam", "$baseDir/tests/input/variant_calling/barcode03/final.bam.bai"])
                input[1] = "final"
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                input[3] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta.fai")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }

    test("Should output three files") {

        when {
            params {
                variant_calling = true 
                variant_caller = "lofreq"
            }
            process {
                """
                input[0] = Channel.of(
                    ["barcode03", "final", "$baseDir/tests/input/variant_calling/barcode03/final.bam", "$baseDir/tests/input/variant_calling/barcode03/final.bam.bai"],
                    ["barcode04", "final", "$baseDir/tests/input/variant_calling/barcode04/final.bam", "$baseDir/tests/input/variant_calling/barcode04/final.bam.bai"],
                    ["barcode05", "final", "$baseDir/tests/input/variant_calling/barcode05/final.bam", "$baseDir/tests/input/variant_calling/barcode05/final.bam.bai"]
                    )
                input[1] = "final"
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                input[3] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta.fai")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
