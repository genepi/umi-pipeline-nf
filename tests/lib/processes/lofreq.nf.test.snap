{
    "Should output three files": {
        "content": [
            {
                "0": [
                    "final.vcf:md5,8761e5c55f8e722dd33472968f33ecd6",
                    "final.vcf:md5,b48b1996596787daa8a83dc973120006",
                    "final.vcf:md5,de4ad8ac00fddc259406af289d16961b"
                ],
                "variants": [
                    "final.vcf:md5,8761e5c55f8e722dd33472968f33ecd6",
                    "final.vcf:md5,b48b1996596787daa8a83dc973120006",
                    "final.vcf:md5,de4ad8ac00fddc259406af289d16961b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.05.0"
        },
        "timestamp": "2024-06-20T09:55:30.290847"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "final.vcf:md5,b48b1996596787daa8a83dc973120006"
                ],
                "variants": [
                    "final.vcf:md5,b48b1996596787daa8a83dc973120006"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.05.0"
        },
        "timestamp": "2024-06-20T09:55:18.706674"
    }
}